tests/cxx/typedef_in_template.cc

```diff
@@ -7,51 +7,54 @@
 //
 //===----------------------------------------------------------------------===//
 
-#include "tests/cxx/typedef_in_template-d1.h"
-
-template<class T>
-class Container {
- public:
-  // Should not be an iwyu violation for T
-  typedef T value_type;
-
-  // C++11 alias declaration, should not be an iwyu violation for T
-  using alias_type = T;
-
-  // IWYU: Pair is...*typedef_in_template-i1.h
```

Comment:
I think we lost coverage for this example. The reason this was interesting was because it proves that the template is not exempt from analysis -- the instantiator of this template should not be responsible for `Pair`, it's an implementation detail of the template.

Reply:
I've restored the deleted test cases.